### PR TITLE
Function for price interpolation

### PR DESCRIPTION
**Pull request by matt-denni**

### CODE REVIEW

**Comment by matt-denni** [comment]
@datawookie I still need to create the tests for this, but here is the function thus far.

**Comment by matt-denni** [comment]
@datawookie I have added the tests for this now.

File: R/price_complete.R [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+#' @return Interpolated prices for the specified (or default) date range as a \code{data.frame}.
+#' @export
+#'
+#' @examples
+#' # Interpolate prices for a specific product within a given date range.
+#' \dontrun{
+#' price_complete(product_id = 1, date_min = "2020-01-01", date_max = "2020-12-31")
+#' }
+price_complete <- function(product_id, date_min = NULL, date_max = NULL) {
+
```
Comment (datawookie) [inline R/price_complete.R:16]: I'm not a big fan of gratuitous whitespace!

File: R/price_complete.R [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  if (!is.null(date_min)) {
+    date_min_check <- try(as.Date(date_min, format="%Y-%m-%d"))
```
Comment (datawookie) [inline R/price_complete.R:18]: You end up calling `as.Date()` twice for each date. Surely you could set `date_min` directly here?

Same applies for `date_max`.

File: R/price_complete.R [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    if("try-error" %in% class(date_max_check) || is.na(date_max_check)) {
+      stop("Argument date_max must be a character string with format YYYY-mm-dd.")
+    }
+  }
+
+  df <- product_prices(product_id) %>%
+    mutate(date = as.Date(time))
+
+  if (is.null(date_min)) {
+    date_start <- min(df$date)
```
Comment (datawookie) [inline R/price_complete.R:35]: To reduce number of variables, you could probably dispense with `date_start` and just use `date_min`.

Sames applies to `date_end` and `date_max`.

File: R/price_complete.R [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    date_end <- max(df$date)
+  } else {
+    date_end <- as.Date(date_max)
+  }
+
+  dates <- seq.Date(date_start, date_end, by = "day")
+
+  complete(df, date = dates, nesting(product_id)) %>%
+    group_by(product_id) %>%
+    mutate(price = approx(date, price, date, method = "constant", rule = 2)$y,
```
Comment (datawookie) [inline R/price_complete.R:50]: I prefer it if you split this and have the assignments for `price` and `is_interpolated` on separate lines.

**Comment by matt-denni** [comment]
@datawookie thanks for the review.  I have made those changes!

**Comment by datawookie** [comment]
@matt-denni this is great. Will you please test this new feature out in a few blog posts. Once we are satisfied that it's production ready then we can merge it into `master`.

**Pull request merged**